### PR TITLE
Support {defaultFn: 'shortid'}

### PR DESCRIPTION
**Pull request by timdp**
[shortid](https://www.npmjs.com/package/shortid) is a pretty popular lightweight alternative to GUIDs. This patch lets you specify `'shortid'` for `defaultFn` so that you don't need to set `default` to `shortid.generate` yourself.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: test/manipulation.test.js [hunk 1]
```diff
@@ -1661,6 +1662,25 @@ describe('manipulation', function() {
...
+
+      before(function(done) {
+        CustomModel = db.define('CustomModel6', {
+          shortid: {type: String, defaultFn: 'shortid'},
+        });
+        db.automigrate('CustomModel6', done);
+      });
+
+      it('should generate a new id when "defaultFn" is "shortid"', function(done) {
+        var inst = CustomModel.create(function(err, m) {
```
Comment (superkhau) [inline test/manipulation.test.js]: Remove `var inst =`

File: test/manipulation.test.js [hunk 2]
```diff
@@ -1661,6 +1662,25 @@ describe('manipulation', function() {
...
+          should.not.exists(err);
```
Comment (superkhau) [inline test/manipulation.test.js]: `if (err) return done(err);`

File: test/manipulation.test.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ var db, Person;
 var ValidationError = require('..').ValidationError;
 
 var UUID_REGEXP = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
+var SHORTID_REGEXP = /^[0-9a-z_\-]{7,14}$/i;
```
Comment (superkhau) [inline test/manipulation.test.js]: Move this directly into the shortid test since it's only used in one place ATM.

File: test/manipulation.test.js (hunk 1 above)
Comment (superkhau) [inline test/manipulation.test.js]: Also change `m` to `model` for clarity

**Comment by timdp** [comment]
@superkhau Happy to make those changes but I based my test off of the ones for GUIDs. Should I update those as well then?

**Comment by superkhau** [comment]
@timdp Let's do this PR properly (put in the fixes I suggested) and then you can create another PR for the refactoring of the other ones (ping me in that PR and I'll help review that one too).

**Comment by timdp** [comment]
@superkhau All right, I'll patch this one first.

**Comment by timdp** [comment]
Added b8ae2c7 and rebased.

**Comment by superkhau** [comment]
test please

**Comment by timdp** [comment]
Is it up to me to somehow fix those failures or is CI broken?

File: test/manipulation.test.js [hunk 4]
```diff
@@ -1703,6 +1703,27 @@ describe('manipulation', function() {
       });
     });
 
+    describe('shortid defaultFn', function() {
+      var CustomModel;
+
+      before(function(done) {
+        CustomModel = db.define('CustomModel6', {
```
Comment (bajtos) [inline test/manipulation.test.js]: Nitpick: can we use a more descriptive name, e.g. `ModelWithShortId`?

Comment (timdp) [inline test/manipulation.test.js]: Okay, added.

**Comment by superkhau** [comment]
>Is it up to me to somehow fix those failures or is CI broken?

No, we're having some CI issues on our end so we look at each issue case by case. Lemme run it again, you can rebase and push again though because juggler is a fast moving target.

**Comment by superkhau** [comment]
test please

File: test/manipulation.test.js [hunk 5]
```diff
@@ -1703,6 +1703,27 @@ describe('manipulation', function() {
       });
     });
 
+    describe('shortid defaultFn', function() {
+      var ModelWithShortid;
+
+      before(function(done) {
+        ModelWithShortid = db.define('ModelWithShortid', {
```
Comment (superkhau) [inline test/manipulation.test.js]: Capital I for id

File: test/manipulation.test.js [hunk 6]
```diff
@@ -1675,7 +1675,7 @@ describe('manipulation', function() {
         db.automigrate('CustomModel5', done);
       });
 
-      it('should generate a new id when "defaultfn" is "uuid"', function(done) {
+      it('should generate a new id when "defaultFn" is "uuid"', function(done) {
```
Comment (superkhau) [inline test/manipulation.test.js]: can you change the capital F back to lowercase f -- to remove this unrelated change.

File: test/manipulation.test.js [hunk 7]
```diff
@@ -1694,7 +1694,7 @@ describe('manipulation', function() {
         db.automigrate('CustomModel5', done);
       });
 
-      it('should generate a new id when "defaultfn" is "uuidv4"', function(done) {
+      it('should generate a new id when "defaultFn" is "uuidv4"', function(done) {
```
Comment (superkhau) [inline test/manipulation.test.js]: same here

**Comment by superkhau** [comment]
Closed in favour of #1107 -- Fixed up the comments I left in this PR and landed. @timdp Thanks for the contribution! ;)

**Comment by slnode** [comment]
Can one of the admins verify this patch?
